modules/configuring-cert-based-auth-quay-cloudsql.adoc

```diff
@@ -0,0 +1,195 @@
+WARNING: psql major version 15, server major version 16.
+         Some psql features might not work.
+SSL connection (protocol: TLSv1.3, cipher: TLS_AES_256_GCM_SHA384, compression: off)
+You are now connected to database "<database_name>" as user "<cloudsql_username>".
+----
+
+.. Add the `pg_trgm` extension if it does not exist by running the following command:
++
+[source,terminal]
+----
+<new_database_name>=> CREATE EXTENSION IF NOT EXISTS pg_trgm;
+----
++
+.Example output
++
+[source,terminal]
+----
+CREATE EXTENSION
+----
+
+.. Sever the connection to your CloudSQL database:
++
+[source,terminal]
+----
+<new_database_name>=> \q 
+----
+
+. In your {productname} directory, create a new YAML file, for example, `postgresql-client-certs.yaml`, by running the following command:
++
+[source,terminal]
+----
+$ touch cloudsql-client-certs
+----
+
+. Create the `postgresql-client-certs` resource by entering the following command:
++
+[source,terminal]
+----
+$ oc create secret generic postgresql-client-certs \
+--from-file config.yaml=<path/to/config.yaml> <1>
+--from-file=tls.crt=<path/to/ssl_client_certificate.pem> <2>
+--from-file=tls.key=<path/to/ssl_client_key.pem> <3>
+--from-file=ca.crt=<path/to/ssl_server_certificate.pem> <4>
+----
+<1> Where` <config.yaml>` is your `base64 decoded` `config.yaml` file.
+<2> Where `ssl_client_certificate.pem` is your SSL certificate in `.pem` format.
+<3> Where `ssl_client_key.pem` is your SSL key in `.pem` format.
+<4> Where `ssl_server_certificate.pem` is your SSL root CA in `.pem` format.
+
+. Depending on your security requirements and the type of certificates that you are using, you can configure {productname} to verify the certificate against a trusted CA using the `verify-ca` SSL mode. Alternatively, you can configure {productname} to verify the certificate against a trusted CA *and* that the hostname matches a SAN entry in your certificates. Use one of the following verification methods.
+
+.. To verify the certificate against a trusted CA only, use the `verify-ca` SSL mode. Edit your `config.yaml` file to include the following information:
++
+[IMPORTANT]
+====
+The information included in the `DB_CONNECTION_ARGS` variable, for example, `sslmode`, `sslrootcert`, `sslcert`, and `sslkey` *must* match the information appended to the `DB_URI` variable. Failure to match might result in a failed connection.
+====
++
+[source,yaml]
+----
+DB_CONNECTION_ARGS:
+    autorollback: true
+    sslmode: verify-ca <1>
+    sslrootcert: /.postgresql/<root.crt> <2>
+    sslcert: /.postgresql/<cloudsql.crt> <3>
+    sslkey: /.postgresql/<cloudsql.key> <4>
+    threadlocals: true <5>
+DB_URI: postgresql://<dbusername>:<dbpassword>@34.29.130.58:5432/quay14?sslmode=verify-ca&sslcert=/.postgresql/<cloudsql.crt>&sslkey=/.postgresql/<cloudsql.key>&sslrootcert=/.postgresql/<root.crt> <6>
+----
+<1> Ensures that the database only tries an SSL/TLS connection and verifies that the server certificate is issued by a trusted CA.
```

Comment:
no, both trusted CA and self-signed CA can work with `verify-ca`, if the hostname matches a SAN entry in your server certificate, then can use `verify-full`.